### PR TITLE
Split headers, etc.

### PR DESCRIPTION
**Pull request by rpavlik**
This is a work in progress but nearly done.

Current state: produces these headers: 
[openxr.zip](https://github.com/KhronosGroup/OpenXR-Hpp/files/5727040/openxr.zip)

cc @matthieucoquet - I incorporated some of your suggestions in here.

### CODE REVIEW

**Comment by rpavlik** [comment]
OK, this is a fairly substantial update, and really makes things better I think. Things are split into individual files to reduce include overhead and let people use e.g. the dispatch without too much else, orthogonality is improved, doxygen is much better, usability bits smoothed out (I ported hello_xr and another project), etc.

Here are the generated headers:
[openxr-hpp-headers.zip](https://github.com/KhronosGroup/OpenXR-Hpp/files/5789777/openxr-hpp-headers.zip)

Here are the doxygen docs:

[html.zip](https://github.com/KhronosGroup/OpenXR-Hpp/files/5789784/html.zip)

File: cmake/FindOpenXR.cmake [hunk 1]
```diff
@@ -1,10 +1,11 @@
-# Copyright 2019 Collabora, Ltd.
+# Copyright 2019-2020 Collabora, Ltd.
```
Comment (jherico) [inline cmake/FindOpenXR.cmake:1]:
```suggestion
# Copyright 2019-2021 Collabora, Ltd.
```

File: cmake/FindOpenXR.cmake [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 # Distributed under the Boost Software License, Version 1.0.
 # (See accompanying file LICENSE_1_0.txt or copy at
 # http://www.boost.org/LICENSE_1_0.txt)
+# SPDX-License-Identifier: BSL-1.0
 #
 # Original Author:
-# 2019 Ryan Pavlik <ryan.pavlik@collabora.com>
+# 2019-2020 Ryan Pavlik <ryan.pavlik@collabora.com>
```
Comment (jherico) [inline cmake/FindOpenXR.cmake:8]:
```suggestion
# 2019-2021 Ryan Pavlik <ryan.pavlik@collabora.com>
```

**Comment by matthieucoquet** [comment]
I tried the headers (with some small changes related to what I mention in https://github.com/KhronosGroup/OpenXR-Hpp/issues/27) and everything works well in my small app.

I just had to download the latest PowerShell to generate the headers. The version by default in Windows results in unclear errors. Maybe it should be mention in the Readme.

The macro to default the dispatcher are pretty nice.

**Comment by rpavlik** [comment]
Great, thanks for the review! I'll make a note in the README

**Comment by rpavlik** [comment]
I actually was able to just modify the script slightly so it works with the standard powershell in Windows 10 :)

**Comment by rpavlik** [comment]
OK, this has been updated slightly. (Main changes are fixing a flags bug, and doing a better job with Bool32.)

Also, for your review pleasure, I have ported hello_xr to the current state of this repo: https://github.com/rpavlik/hello_xr_hpp/tree/pr26-210112

Each commit is buildable so you can see the process I used.

**Pull request merged**